packages/@react-spectrum/toast/src/ToastContainer.tsx

```diff
@@ -10,43 +10,179 @@
  * governing permissions and limitations under the License.
  */
 
-import {classNames} from '@react-spectrum/utils';
-import React, {ReactElement} from 'react';
-import {Toast} from './';
-import toastContainerStyles from './toastContainer.css';
-import {ToastState} from '@react-types/toast';
-// import {useProvider} from '@react-spectrum/provider';
-
-export function ToastContainer(props: ToastState): ReactElement {
-  let {
-    onRemove,
-    toasts
-  } = props;
-  // let providerProps = useProvider();
-  let toastPlacement = 'bottom'; /* providerProps && providerProps.toastPlacement && providerProps.toastPlacement.split(' '); */
-  let containerPosition = toastPlacement && toastPlacement[0];
-  let containerPlacement = toastPlacement && toastPlacement[1];
-
-  let renderToasts = () => toasts.map((toast) =>
-    (<Toast
-      {...toast.props}
-      key={toast.props.toastKey}
-      onRemove={onRemove}
-      timer={toast.timer}>
-      {toast.content}
-    </Toast>)
-  );
-
-
-  return (
-    <div
-      className={classNames(
-        toastContainerStyles,
-        'react-spectrum-ToastContainer',
-        containerPosition && `react-spectrum-ToastContainer--${containerPosition}`,
-        containerPlacement && `react-spectrum-ToastContainer--${containerPlacement}`
-      )}>
-      {renderToasts()}
-    </div>
-  );
+import {AriaToastRegionProps} from '@react-aria/toast';
+import React, {ReactElement, ReactNode, useEffect, useRef} from 'react';
+import {SpectrumToastValue, Toast} from './Toast';
+import {Toaster} from './Toaster';
+import {ToastOptions, ToastQueue, useToastQueue} from '@react-stately/toast';
+import {useSyncExternalStore} from 'use-sync-external-store/shim';
```

Comment:
Can you add a comment about why we are using this shim? Either here, in its usage or in the stately hook

Reply:
Added in stately.